src/devices/Pca95x4/Pca95x4.cs

```diff
@@ -0,0 +1,199 @@
+            {
+                throw new Exception("Master controller has not been initialized.");
+            }
+
+            if (_interrupt == null)
+            {
+                throw new Exception("INT pin has not been initialized.");
+            }
+
+            return _masterGpioController.Read((int)_interrupt);
+        }
+
+        /// <summary>
+        /// Writes a byte to a register.
+        /// </summary>
+        /// <param name="register">The register to write.</param>
+        /// <param name="data">The data to write to the register.</param>
+        public void Write(Register register, byte data)
+        {
+            Write(register, new byte[] { data });
+        }
+
+        /// <summary>
+        /// Writes a number of bytes to a register.
+        /// All data will be written to selected register as Pca95x4 does not auto-increment addressing.
+        /// </summary>
+        /// <param name="register">The register to write.</param>
+        /// <param name="data">The data to write to the register.</param>
+        public void Write(Register register, byte[] data)
+        {
+            byte[] writeBuffer = new byte[data.Length + 1]; // Include Register Address.
+            writeBuffer[0] = (byte)register;
+            data.CopyTo(writeBuffer, 1);
+            _i2cDevice.Write(writeBuffer);
+        }
+
+        /// <summary>
+        /// Writes to a register bit.
+        /// </summary>
+        /// <param name="register">The register to write.</param>
+        /// <param name="bitNumber">The register bit number to write.</param>
+        /// <param name="bit">The value to write to register bit.</param>
+        public void WriteBit(Register register, int bitNumber, bool bit)
+        {
+            byte data = Read(register);
+
+            if (bit)
+            {
+                SetBit(ref data, bitNumber);
+            }
+            else
+            {
+                ClearBit(ref data, bitNumber);
+            }
+
+            Write(register, data);
+        }
+
+        /// <summary>
+        /// Inverts Input Register polarity.
+        /// </summary>
+        /// <param name="invert">Determines if Inputer Register polarity is inverted.</param>
+        public void InvertInputRegisterPolarity(bool invert)
+        {
+            byte data = invert ? (byte)0xFF : (byte)0x00;
+            Write(Register.PolarityInversion, data);
+        }
+
+        /// <summary>
+        /// Inverts Input Register bit polarity.
+        /// </summary>
+        /// <param name="bitNumber">The Input Register bit number to invert.</param>
+        /// <param name="invert">Determines if the Input Register bit polarity is inverted.</param>
+        public void InvertInputRegisterBitPolarity(int bitNumber, bool invert)
+        {
+            WriteBit(Register.PolarityInversion, bitNumber, invert);
+        }
+
+        public void Dispose()
+        {
+            _i2cDevice?.Dispose();
```

Comment:
NIT: you should probably set this guy to null after you dispose it in order to avoid potential issues with double dispose.

Reply:
I noticed we didn't null out the I2cDevice and SpiDevice in Mcp23xxx.  I'll try to do this at some point.